Changed the physics, specie, and geometry interfaces and all dependen…

vflorins opened the pull request:
This commit contains change to the physics, specie, and geometry interfaces in anticipation of the "big change" by Lucius. The servers now use units from the `Fluid` namespace, and the rest of the particle codes use the `Particle` units. The benchmarks files were corrected and the test compile and execute, although the results are not always correct because of differences in normalization factors. The build system was also modified; `configure.ac` now takes different options such as `--with_cuda= `, and `--enable_gsl` (GSL is now optional). It also exports conditional variables to the makefiles which don't need custom library and include variables anymore.

vflorins committed:
Changed the physics, specie, and geometry interfaces and all dependent files

jgaloguz reviewed (changes requested):
The benchmark tests in vf_temp1 don't run. They fail at the compilation stage due to a missing file `common/coordinates.hh`.

vflorins committed:
Add files via upload

Added the missing geometry.hh

vflorins commented:
Please test again; the missing file (`coordinates.hh`) was uploaded.

jgaloguz reviewed (changes requested):
The coordinates dependency error went away, but the SERIAL tests don't run because they're not compiled with MPI but `server_config.hh` includes `server_base.hh` which has some MPI structures. We should revise our file guards/macros.

vflorins committed:
Disabled MPI specific headers when building serial code

vflorins commented:
Edited `server_config.hh` to remove MPI includes in `SERIAL` mode. Also modified `specie.hh` as requested.

jgaloguz committed:
Updated a few of the benchmark tests so they would run.
The benchmarks in benchmarks/main_test_parker_spiral.cc and benchmarks/main_test_init_cond_records.cc simply needed to be printed out in physical units. In particular, the radius of initial positions in the second code was changed from 1 au to the number 1, since the benchmark only intends to test the initial conditions capabilities (records and MPI communication of distributions objects), so the physical extent of the initial positions is not tied to a physical quantity.
The test benchmarks/main_test_dipole_periods.cc should also be printed in physical units (plotting script updated accordingly), but the physics are still messed up in the Guiding Center transport module. We fixed this yesterday on Dr Florinski's local version, along with some other bugs we discovered in the Lorentz integrator.
The run_tests.sh script was enhanced to receive input from the user specifying which test to run. The current options are running 1 specified benchmark test or all of them in series.
Benchmarks 10 and 11 are still broken. We found some potential issues with the interaction between fluid and particle units in the server background, which could be causing the errors in test 11, but not in 10.

jgaloguz commented:
I just pushed the fixes to a couple of benchmarks that pretty much only needed plotting changes (see commit for details). Please pull before editing this branch to push fixes for the others.

jgaloguz committed:
Fixed configure.ac
The PARKER_SOURCE transport type was removed from configure.ac, which was causing the diffusive shock acceleration benchmark test to fail. Once this trajectory option was added the code worked.

jgaloguz committed:
Update mpi_config.cc

Fixed a comment

vflorins committed:
Minor fixes for select background and trajectory classes

vflorins committed:
Updated file in /runs

jgaloguz committed:
Fixed configure.ac (again...)
configure.ac had the server_interp_order and num_ghost_cells options deleted during the Nov 15, 2025 commit. **PLEASE pay more attention when removing blocks of code and test all changes thoroughly in an incremental fashion.** This was causing (silent) errors when attempting to use the server background types. These options have been added to the configuration script.
In addition, the new compilation options threw a lot more warnings than before (eg. int vs long unsigned int comparison). Some files were edited to remove these warnings, although some warnings persist in parts of the code that are still in development (MHD stuff).
There were a few orphaned c_code references in server_cartesian.cc that we missed after the original push to the new units convention. These now belong to the Fluid units.
All benchmark tests run successfully. The parker spiral magnetic field test was enhanced to also compare with the discretized background version. The run_tests.sh script was improved by adding timing to the run portions of the tests. Also, the tests have been reorganized in order of CPUs required to run. As a point of comparison, script ran for all tests and 8 Intel i7-12700KF CPUs (including configuration, compilation, and plotting) took just under 9 minutes to complete. The longest runtime for a single test was 109 seconds for the PITCH ANGLE DISTRIBUTION ISOTROPIZATION benchmark.

jgaloguz committed:
Minor change to run_tests.sh in benchmarks folder

jgaloguz commented:
All the benchmarks run successfully now. The problem with the server backgrounds was that the original configure.ac edit in this pull request DELETED the interpolation order and number of ghost cells options so the corresponding macros were undefined and the code was defaulting to zeroth order interpolation and no ghost cells. Please, run the benchmarks in your local machines and/or the cluster and confirm they run successfully on your end before merging.

vflorins commented:
Confirmed that the tests run correctly.
